src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -12,8 +12,46 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
+    /// <summary>
+    /// <para>
+    /// A utility class for making a decision dag for a pattern-matching construct. A decision dag is represented by
+    /// the class <see cref="BoundDecisionDag"/> and is a representation of a finite state automaton that performs a
+    /// sequence of binary tests. Each node is represented by a <see cref="BoundDecisionDagNode"/>. There are four
+    /// kind of nodes: <see cref="BoundTestDecisionDagNode"/> performs one of the binary tests;
+    /// <see cref="BoundEvaluationDecisionDagNode"/> simply performs some computation and stores it in one or more
+    /// temporary variables for use in subsequent nodes (think of it as a node with a single successor);
+    /// <see cref="BoundWhenDecisionDagNode"/> represents the test performed by evaluating the expression of the
+    /// when-clause of a switch case; and <see cref="BoundLeafDecisionDagNode"/> represents a leaf node when we
+    /// have finally determined exactly which case matches. Each test processes a single input, and there are
+    /// four kinds:<see cref="BoundDagNullTest"/> tests a value for null; <see cref="BoundDagNonNullTest"/>
+    /// tests that a value is not null; <see cref="BoundDagTypeTest"/> checks if the value is of a given type;
+    /// and <see cref="BoundDagValueTest"/> checks if the value is equal to a given constant. Of the evaluations,
+    /// there are <see cref="BoundDagDeconstructEvaluation"/> which represents an invocation of a type's
+    /// "Deconstruct" method; <see cref="BoundDagFieldEvaluation"/> reads a field; <see cref="BoundDagPropertyEvaluation"/>
+    /// reads a property; and <see cref="BoundDagTypeEvaluation"/> converts a value from one type to another (which
```

Comment:
>BoundDagTypeEvaluation [](start = 41, length = 22)

Would it make sense to call this `BoundDagCastEvaluation`?

Reply:
Well, it only converts a value to a type that it already has at runtime. A cast does so much more.